### ISSUE TITLE
False positives on table line length when table not surrounded by whitespace

### ISSUE THREAD

**Issue by markeganfuller** [markdownlint/markdownlint#154]
If a table is not surrounded by whitespace, eg:

```markdown
# Document

| abc | def | ghi |
| --- | --- | --- |
| abc | this line is too long but thats okay beacause we've turned the rule off for tables right? | ghi |        
bob
```

And we use the following ruleset:

```ruby
all
rule 'MD013', :tables => false
```

We get the following warnings as the table is not correctly recognised: 

```bash
$ mdl --version
0.4.0
$ mdl test.md -s mdl_style.rb
test.md:5: MD013 Line length
```

If we 'correct' the syntax for the table
then the table is correctly recognised and the ignore works. 

Due to the false positive and the effect of having no whitespace around the table on parsers
is it worth having a rule for whitespace around tables similar to these existing rules?

* [MD022 - Headers should be surrounded by blank lines](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md#md022---headers-should-be-surrounded-by-blank-lines)
* [MD031 - Fenced code blocks should be surrounded by blank lines](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md#md031---fenced-code-blocks-should-be-surrounded-by-blank-lines)
* [MD032 - Lists should be surrounded by blank lines](https://github.com/mivok/markdownlint/blob/master/docs/RULES.md#md032---lists-should-be-surrounded-by-blank-lines)

**Comment by Prome88** [comment]
@markeganfuller how did you "correct" the table syntax to avoid this false positve?

**Comment by markeganfuller** [comment]
You need to ensure the table is surrounded by whitespace, just like headers, fenced code blocks and lists.

Broken:
```
# Document

| abc | def | ghi |
| --- | --- | --- |
| abc | this line is too long but thats okay beacause we've turned the rule off for tables right? | ghi |        
bob
```

Fixed:

```
# Document

| abc | def | ghi |
| --- | --- | --- |
| abc | this line is too long but thats okay beacause we've turned the rule off for tables right? | ghi |        

bob
```
